common-libs/internal-api-clients/src/services/DidAuthApiService.ts

```diff
@@ -0,0 +1,60 @@
+import { profile } from '@affinidi/common'
+import { DidAuthService } from '@affinidi/affinidi-did-auth-lib'
+
+import { RawApiSpec, RequestOptionsForOperation } from '../types/request'
+import { BuiltApiType } from '../types/typeBuilder'
+import GenericApiService, { GenericConstructorOptions } from './GenericApiService'
+
+export type DidAuthConstructorOptions = GenericConstructorOptions & {
+  audienceDid: string
+  didAuthService: DidAuthService
+}
+@profile()
+export default abstract class DidAuthApiService<
+  TApi extends BuiltApiType,
+  TDidAuthOperationId extends keyof TApi
```

Comment:
We should only allow operations that are compatible with what we expect did auth operation to look like.

So it should be not `keyof TApi` but rather only keys such that `TApi[key]['response']` extends the expected response, expected request extends `TApi[key]['request']`, etc.

Reply:
You mean: we need to filter out only the endpoints that have did-auth from the `TApi` definition? E.g. only endpoints that require `bearerAuth` security?

Reply:
I mean, we need to filter out only the endpoints that accept `{ audienceString: string }` and return `string`. So that you'll have typechecking in `_createDidAuthToken`

Reply:
I've changed this to only accept operation ids that look like create did auth requests. However, the inner method still has type casting. I don't mind this and since the public API is better than before, and writing more TS would only change internals of this class, I would prefer not to spend more time on this and don't want to complicate the types.

But If you think we should still improve the types, I would be happy to pair and try to improve it. LMK what you think.